File: Ruby/response/payment/AmountReservationTransaction.rb

class AmountReservationTransaction

	def initialize
		@clientCorrelator=nil
		@endUserId=nil
		@paymentAmount=nil
		@referenceCode=nil
		@referenceSequence=0
		@resourceURL=nil
		@serverReferenceCode=nil
		@transactionStatus=nil
	end

	def initializeJSON(jsondict)
		@clientCorrelator=nil
		if (jsondict!=nil) && (jsondict.has_key?'clientCorrelator') && (jsondict['clientCorrelator']!=nil)
			@clientCorrelator=jsondict['clientCorrelator']
		end
		@endUserId=nil
		if (jsondict!=nil) && (jsondict.has_key?'endUserId') && (jsondict['endUserId']!=nil)
			@endUserId=jsondict['endUserId']
		end
		@paymentAmount=nil
		if (jsondict!=nil) && (jsondict.has_key?'paymentAmount') && (jsondict['paymentAmount']!=nil) then
			@paymentAmount=PaymentAmount.new
			@paymentAmount.initializeJSON(jsondict['paymentAmount'])
		end
		@referenceCode=nil
		if (jsondict!=nil) && (jsondict.has_key?'referenceCode') && (jsondict['referenceCode']!=nil)
			@referenceCode=jsondict['referenceCode']
		end
		@referenceSequence=0
		if (jsondict!=nil) && (jsondict.has_key?'referenceSequence') && (jsondict['referenceSequence']!=nil)
			@referenceSequence=jsondict['referenceSequence'].to_i
		end
		@resourceURL=nil
		if (jsondict!=nil) && (jsondict.has_key?'resourceURL') && (jsondict['resourceURL']!=nil)
			@resourceURL=jsondict['resourceURL']
		end
		@serverReferenceCode=nil
		if (jsondict!=nil) && (jsondict.has_key?'serverReferenceCode') && (jsondict['serverReferenceCode']!=nil)
			@serverReferenceCode=jsondict['serverReferenceCode']
		end
		@transactionStatus=nil
		if (jsondict!=nil) && (jsondict.has_key?'transactionStatus') && (jsondict['transactionStatus']!=nil)
			@transactionStatus=jsondict['transactionStatus']
		end
	end

	def getClientCorrelator
		@clientCorrelator
	end

	def setClientCorrelator(clientCorrelator)
		@clientCorrelator=clientCorrelator
	end

	def getEndUserId
		@endUserId
	end

	def setEndUserId(endUserId)
		@endUserId=endUserId
	end

	def getPaymentAmount
		@paymentAmount
	end

	def setPaymentAmount(paymentAmount)
		@paymentAmount=paymentAmount
	end

	def getReferenceCode
		@referenceCode
	end

	def setReferenceCode(referenceCode)
		@referenceCode=referenceCode
	end

	def getReferenceSequence
		@referenceSequence
	end

	def setReferenceSequence(referenceSequence)
		@referenceSequence=referenceSequence
	end

	def getResourceURL
		@resourceURL
	end

	def setResourceURL(resourceURL)
		@resourceURL=resourceURL
	end

	def getServerReferenceCode
		@serverReferenceCode
	end

	def setServerReferenceCode(serverReferenceCode)
		@serverReferenceCode=serverReferenceCode
	end

	def getTransactionStatus
		@transactionStatus
	end

	def setTransactionStatus(transactionStatus)
		@transactionStatus=transactionStatus
	end


end 

File: Python/response/payment/AmountReservationTransaction.py

class AmountReservationTransaction:
	"""class containing the payment details response received from the OneAPI server when a more complex payment reservation type transaction is applied to the mobile user account"""
	
	def __init__(self):
		"""Default class constructor"""
		self.clientCorrelator=None
		self.endUserId=None
		self.paymentAmount=None
		self.referenceCode=None
		self.referenceSequence=0
		self.resourceURL=None
		self.serverReferenceCode=None
		self.transactionStatus=None
	
	def __init__(self, jsondict):
		"""Class constructor that will create an instance initialised from a parsed JSON data block"""
		self.clientCorrelator=None
		if jsondict is not None and 'clientCorrelator' in jsondict and jsondict['clientCorrelator'] is not None:
			self.clientCorrelator=jsondict['clientCorrelator']
		self.endUserId=None
		if jsondict is not None and 'endUserId' in jsondict and jsondict['endUserId'] is not None:
			self.endUserId=jsondict['endUserId']
		self.paymentAmount=None
		if jsondict is not None and 'paymentAmount' in jsondict and jsondict['paymentAmount'] is not None:
			self.paymentAmount=PaymentAmount(jsondict['paymentAmount'])
		self.referenceCode=None
		if jsondict is not None and 'referenceCode' in jsondict and jsondict['referenceCode'] is not None:
			self.referenceCode=jsondict['referenceCode']
		self.referenceSequence=0
		if jsondict is not None and 'referenceSequence' in jsondict and jsondict['referenceSequence'] is not None:
			self.referenceSequence=int( jsondict['referenceSequence'])
		self.resourceURL=None
		if jsondict is not None and 'resourceURL' in jsondict and jsondict['resourceURL'] is not None:
			self.resourceURL=jsondict['resourceURL']
		self.serverReferenceCode=None
		if jsondict is not None and 'serverReferenceCode' in jsondict and jsondict['serverReferenceCode'] is not None:
			self.serverReferenceCode=jsondict['serverReferenceCode']
		self.transactionStatus=None
		if jsondict is not None and 'transactionStatus' in jsondict and jsondict['transactionStatus'] is not None:
			self.transactionStatus=jsondict['transactionStatus']
	
	def getClientCorrelator(self):
		"""getter for clientCorrelator : uniquely identifies this charging request"""
		return self.clientCorrelator
	
	def setClientCorrelator(self,clientCorrelator):
		"""setter for clientCorrelator : uniquely identifies this charging request"""
		self.clientCorrelator=clientCorrelator
	
	def getEndUserId(self):
		"""getter for endUserId : MSISDN or ACR of the mobile user being charged"""
		return self.endUserId
	
	def setEndUserId(self,endUserId):
		"""setter for endUserId : MSISDN or ACR of the mobile user being charged"""
		self.endUserId=endUserId
	
	def getPaymentAmount(self):
		"""getter for paymentAmount : detailed payment information for the charge/ refund"""
		return self.paymentAmount
	
	def setPaymentAmount(self,paymentAmount):
		"""setter for paymentAmount : detailed payment information for the charge/ refund"""
		self.paymentAmount=paymentAmount
	
	def getReferenceCode(self):
		"""getter for referenceCode : reference (supplier) for this request"""
		return self.referenceCode
	
	def setReferenceCode(self,referenceCode):
		"""setter for referenceCode : reference (supplier) for this request"""
		self.referenceCode=referenceCode
	
	def getReferenceSequence(self):
		"""getter for referenceSequence : reference sequence for the charging operation (starting from 1)"""
		return self.referenceSequence
	
	def setReferenceSequence(self,referenceSequence):
		"""setter for referenceSequence : reference sequence for the charging operation (starting from 1)"""
		self.referenceSequence=referenceSequence
	
	def getResourceURL(self):
		"""getter for resourceURL : specifies the URL to this charging request on the server"""
		return self.resourceURL
	
	def setResourceURL(self,resourceURL):
		"""setter for resourceURL : specifies the URL to this charging request on the server"""
		self.resourceURL=resourceURL
	
	def getServerReferenceCode(self):
		"""getter for serverReferenceCode : reference (server) for this request"""
		return self.serverReferenceCode
	
	def setServerReferenceCode(self,serverReferenceCode):
		"""setter for serverReferenceCode : reference (server) for this request"""
		self.serverReferenceCode=serverReferenceCode
	
	def getTransactionStatus(self):
		"""getter for transactionStatus : status for this charge/refund transaction"""
		return self.transactionStatus
	
	def setTransactionStatus(self,transactionStatus):
		"""setter for transactionStatus : status for this charge/refund transaction"""
		self.transactionStatus=transactionStatus
	
